### PR TITLE
Fixed #38

### LINKED ISSUE #38: java.lang.NoClassDefFoundError: org/virtuslab/akkasaferserializer/SerializabilityTrait
I will provide a full repro once the legal stuff about your access to Hydra repo is finally sorted out...

```
[info] compiling 73 Scala sources to /home/plipski/hydra-backend/services/target/scala-2.12/classes ...
[info] AkkaSerializabilityCheckerPlugin: Found new annotated trait: hydra.CborSerializable
[error] ## Exception when compiling 337 sources to /home/plipski/hydra-backend/services/target/scala-2.12/classes
[error] java.lang.NoClassDefFoundError: org/virtuslab/akkasaferserializer/SerializabilityTrait
[error] org.virtuslab.akkasaferserializer.AkkaSerializabilityCheckerPluginComponent$$anon$1.$anonfun$apply$2(AkkaSerializabilityCheckerPluginComponent.scala:49)
[error] scala.collection.immutable.List.$anonfun$foldRight$1(List.scala:447)
[error] scala.collection.immutable.List.foldRight(List.scala:91)
[error] org.virtuslab.akkasaferserializer.AkkaSerializabilityCheckerPluginComponent$$anon$1.apply(AkkaSerializabilityCheckerPluginComponent.scala:31)
[error] scala.tools.nsc.Global$GlobalPhase.applyPhase(Global.scala:454)
[error] scala.tools.nsc.Global$GlobalPhase.run(Global.scala:402)
[error] scala.tools.nsc.Global$Run.compileUnitsInternal(Global.scala:1511)
[error] scala.tools.nsc.Global$Run.compileUnits(Global.scala:1495)
[error] scala.tools.nsc.Global$Run.compileSources(Global.scala:1488)
[error] scala.tools.nsc.Global$Run.compileFiles(Global.scala:1596)
[error] xsbt.CachedCompiler0.run(CompilerBridge.scala:163)
[error] xsbt.CachedCompiler0.run(CompilerBridge.scala:134)
[error] xsbt.CompilerBridge.run(CompilerBridge.scala:39)
[error] sbt.internal.inc.AnalyzingCompiler.compile(AnalyzingCompiler.scala:92)
[error] sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$7(MixedAnalyzingCompiler.scala:186)
[error] scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[error] sbt.internal.inc.MixedAnalyzingCompiler.timed(MixedAnalyzingCompiler.scala:241)
[error] sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4(MixedAnalyzingCompiler.scala:176)
[error] sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4$adapted(MixedAnalyzingCompiler.scala:157)
[error] sbt.internal.inc.JarUtils$.withPreviousJar(JarUtils.scala:239)
[error] sbt.internal.inc.MixedAnalyzingCompiler.compileScala$1(MixedAnalyzingCompiler.scala:157)
[error] sbt.internal.inc.MixedAnalyzingCompiler.compile(MixedAnalyzingCompiler.scala:204)
[error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1(IncrementalCompilerImpl.scala:573)
[error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1$adapted(IncrementalCompilerImpl.scala:573)
[error] sbt.internal.inc.Incremental$.$anonfun$apply$5(Incremental.scala:173)
[error] sbt.internal.inc.Incremental$.$anonfun$apply$5$adapted(Incremental.scala:171)
[error] sbt.internal.inc.Incremental$$anon$2.run(Incremental.scala:458)
[error] sbt.internal.inc.IncrementalCommon$CycleState.next(IncrementalCommon.scala:116)
[error] sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:56)
[error] sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:52)
[error] sbt.internal.inc.IncrementalCommon.cycle(IncrementalCommon.scala:261)
[error] sbt.internal.inc.Incremental$.$anonfun$incrementalCompile$8(Incremental.scala:413)
[error] sbt.internal.inc.Incremental$.withClassfileManager(Incremental.scala:498)
[error] sbt.internal.inc.Incremental$.incrementalCompile(Incremental.scala:400)
[error] sbt.internal.inc.Incremental$.apply(Incremental.scala:165)
[error] sbt.internal.inc.IncrementalCompilerImpl.compileInternal(IncrementalCompilerImpl.scala:573)
[error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileIncrementally$1(IncrementalCompilerImpl.scala:491)
[error] sbt.internal.inc.IncrementalCompilerImpl.handleCompilationError(IncrementalCompilerImpl.scala:332)
[error] sbt.internal.inc.IncrementalCompilerImpl.compileIncrementally(IncrementalCompilerImpl.scala:420)
[error] sbt.internal.inc.IncrementalCompilerImpl.compile(IncrementalCompilerImpl.scala:137)
[error] sbt.Defaults$.compileIncrementalTaskImpl(Defaults.scala:2176)
[error] sbt.Defaults$.$anonfun$compileIncrementalTask$2(Defaults.scala:2133)
[error] sbt.internal.io.Retry$.apply(Retry.scala:40)
[error] sbt.internal.io.Retry$.apply(Retry.scala:23)
[error] sbt.internal.server.BspCompileTask$.compute(BspCompileTask.scala:31)
[error] sbt.Defaults$.$anonfun$compileIncrementalTask$1(Defaults.scala:2129)
[error] scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error] sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error] sbt.std.Transform$$anon$4.work(Transform.scala:68)
[error] sbt.Execute.$anonfun$submit$2(Execute.scala:282)
[error] sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
[error] sbt.Execute.work(Execute.scala:291)
[error] sbt.Execute.$anonfun$submit$1(Execute.scala:282)
[error] sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error] sbt.CompletionService$$anon$2.call(CompletionService.scala:64)
[error] java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
[error] java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[error] java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[error] java.base/java.lang.Thread.run(Thread.java:834)
```

### PR DESCRIPTION
**Pull request by MarconZet**

### CODE REVIEW

File: checker-plugin/src/main/scala/org/virtuslab/akkasaferserializer/AkkaSerializabilityCheckerPluginComponent.scala [hunk 1]
```diff
@@ -13,6 +13,8 @@ class AkkaSerializabilityCheckerPluginComponent(val pluginOptions: PluginOptions
 
   override def newPhase(prev: Phase): Phase =
     new StdPhase(prev) {
+      private val serializabilityTraitType = typeOf[SerializabilityTrait]
```
Comment (PawelLipski) [inline checker-plugin/src/main/scala/org/virtuslab/akkasaferserializer/AkkaSerializabilityCheckerPluginComponent.scala:16]: Hmmm so what was the exact cause of `NoClassDefFoundError`? 🤔

File: checker-plugin/src/main/scala/org/virtuslab/akkasaferserializer/AkkaSerializabilityCheckerPluginComponent.scala [hunk 2]
```diff
@@ -46,10 +48,10 @@ class AkkaSerializabilityCheckerPluginComponent(val pluginOptions: PluginOptions
                 case None =>
                   reporter.error(
                     next.typeSymbol.pos,
-                    s"""${next.toString()} is used as Akka message but does not extend a trait annotated with ${classOf[
```
Comment (PawelLipski) [inline checker-plugin/src/main/scala/org/virtuslab/akkasaferserializer/AkkaSerializabilityCheckerPluginComponent.scala:49]: Oh ok... the error was raised here right?

File: checker-plugin/src/test/scala/org/virtuslab/akkasaferserializer/PluginComponentSpec.scala [hunk 3]
```diff
@@ -55,4 +55,9 @@ class PluginComponentSpec extends AnyFlatSpecLike with should.Matchers {
   it should "be able to detect serializer trait in generics" in {
     testCode(getResourceAsString("GenericsTest.scala"))
   }
+
+  it should "detect lack of upper bounds in generics" in {
```
Comment (PawelLipski) [inline checker-plugin/src/test/scala/org/virtuslab/akkasaferserializer/PluginComponentSpec.scala:59]: Does this PR fix both `akka-safer-serializer-failing` and `akka-safer-serializer-exception` branches from Hydra repo? I'm not sure if both were corresponding to the same error or two different ones 🤔

File: checker-plugin/src/main/scala/org/virtuslab/akkasaferserializer/AkkaSerializabilityCheckerPluginComponent.scala (hunk 1 above)
Comment (MarconZet) [reply checker-plugin/src/main/scala/org/virtuslab/akkasaferserializer/AkkaSerializabilityCheckerPluginComponent.scala:16]: https://github.com/sbt/sbt/issues/2255
CheckerPlugin is quite basic, so this solution works.
DumpSchemaPlugin will require one of the workarounds suggested in this issue.

File: checker-plugin/src/test/scala/org/virtuslab/akkasaferserializer/PluginComponentSpec.scala (hunk 3 above)
Comment (MarconZet) [reply checker-plugin/src/test/scala/org/virtuslab/akkasaferserializer/PluginComponentSpec.scala:59]: `sbt compile` reported error on `akka-safer-serializer-failing` but didn't report anything on `akka-safer-serializer-exception`.
Maybe `akka-safer-serializer-exception`  was about some false positive that was already solved?

File: checker-plugin/src/main/scala/org/virtuslab/akkasaferserializer/AkkaSerializabilityCheckerPluginComponent.scala (hunk 2 above)
Comment (MarconZet) [reply checker-plugin/src/main/scala/org/virtuslab/akkasaferserializer/AkkaSerializabilityCheckerPluginComponent.scala:49]: Exactly

**Review by PawelLipski — APPROVED** [review]

**Comment by PawelLipski** [comment]
@MarconZet merge this one?

**Pull request merged**